Tiltfile

```diff
@@ -204,19 +231,34 @@ def enable_provider(name):
 # setup if you're repeatedly destroying and recreating your kind cluster, as it doesn't have to pull the images over
 # the network each time.
 def deploy_cert_manager():
-    registry = "quay.io/jetstack"
-    version = "v0.11.0"
-    images = ["cert-manager-controller", "cert-manager-cainjector", "cert-manager-webhook"]
-
-    if settings.get("preload_images_for_kind"):
-        for image in images:
-            local("docker pull {}/{}:{}".format(registry, image, version))
-            local("kind load docker-image --name {} {}/{}:{}".format(settings.get("kind_cluster_name"), registry, image, version))
-
-    local("kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/{}/cert-manager.yaml".format(version))
-
-    # wait for the service to become available
-    local("kubectl wait --for=condition=Available --timeout=300s apiservice v1beta1.webhook.cert-manager.io")
+    registry = settings.get("cert_manager_registry", "quay.io/jetstack")
+    version = settings.get("cert_manager_version", "v0.16.0")
+
+    # check if cert-mamager is already installed, otherwise pre-load images & apply the manifest
+    # NB. this is required until https://github.com/jetstack/cert-manager/issues/3121 is addressed otherwise
+    # when applying the manifest twice to same cluster kubectl get stuck
+    existsCheck = str(local("kubectl get namespaces"))
+    if existsCheck.find("cert-manager") == -1:
+        # pre-load cert-manager images in kind
+        images = ["cert-manager-controller", "cert-manager-cainjector", "cert-manager-webhook"]
+        if settings.get("preload_images_for_kind"):
+            for image in images:
+                local("docker pull {}/{}:{}".format(registry, image, version))
+                local("kind load docker-image --name {} {}/{}:{}".format(settings.get("kind_cluster_name"), registry, image, version))
+
+        # apply the cert-manager manifest
+        local("kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/{}/cert-manager.yaml".format(version))
```

Comment:
Does this only happen if you already have a cluster up and running with an older version?

Reply:
From my tests this happens when you have a cluster running with v01.6.0 already installed
(tilt up, then exit from tilt without cleaning up your environment, then tilt up again)